src/typeChecker/utils.ts

```diff
@@ -0,0 +1,391 @@
+// =======================================
+// Helper functions/constants for type checker and type error checker
+// =======================================
+
+import {
+  AllowedDeclarations,
+  BindableType,
+  Chapter,
+  ForAll,
+  FunctionType,
+  List,
+  LiteralType,
+  Pair,
+  PredicateType,
+  Primitive,
+  SArray,
+  TSBasicType,
+  Type,
+  TypeEnvironment,
+  UnionType,
+  Variable
+} from '../types'
+import * as tsEs from './tsESTree'
+
+// Name of Unary negative builtin operator
+export const NEGATIVE_OP = '-_1'
+
+// Special name used for saving return type in type environment
+export const RETURN_TYPE_IDENTIFIER = '//RETURN_TYPE'
+
+export const typeAnnotationKeywordToBasicTypeMap = {
+  [tsEs.TSNodeType.TSAnyKeyword]: TSBasicType.ANY,
+  [tsEs.TSNodeType.TSBigIntKeyword]: TSBasicType.BIGINT,
+  [tsEs.TSNodeType.TSBooleanKeyword]: TSBasicType.BOOLEAN,
+  [tsEs.TSNodeType.TSNeverKeyword]: TSBasicType.NEVER,
+  [tsEs.TSNodeType.TSNullKeyword]: TSBasicType.NULL,
+  [tsEs.TSNodeType.TSNumberKeyword]: TSBasicType.NUMBER,
+  [tsEs.TSNodeType.TSObjectKeyword]: TSBasicType.OBJECT,
+  [tsEs.TSNodeType.TSStringKeyword]: TSBasicType.STRING,
+  [tsEs.TSNodeType.TSSymbolKeyword]: TSBasicType.SYMBOL,
+  [tsEs.TSNodeType.TSUndefinedKeyword]: TSBasicType.UNDEFINED,
+  [tsEs.TSNodeType.TSUnknownKeyword]: TSBasicType.UNKNOWN,
+  [tsEs.TSNodeType.TSVoidKeyword]: TSBasicType.VOID
+}
```

Comment:
```suggestion
export const typeAnnotationKeywordToBasicTypeMap: Record<tsEs.TSTypeKeyword, TSBasicType> = {
  'TSAnyKeyword': 'any',
  'TSBigIntKeyword': 'bigint',
  'TSBooleanKeyword': 'boolean',
  'TSNeverKeyword': 'never',
  'TSNullKeyword': 'null',
  'TSNumberKeyword': 'number',
  'TSObjectKeyword': 'object',
  'TSStringKeyword': 'string',
  'TSSymbolKeyword': 'symbol',
  'TSUndefinedKeyword': 'undefined',
  'TSUnknownKeyword': 'unknown',
  'TSVoidKeyword': 'void'
}
```

A benefit of using union types over enums is that your typing can be tighter here. Instead of using `TSNodeType` which is a union type of your 3 enums `TSTypeAnnotationType | TSTypeKeyword | TSOnlyNode`, you can use `TSTypeKeyword`.